firestore/collgroupref.go

```diff
@@ -54,6 +54,9 @@ func newCollectionGroupRef(c *Client, dbPath, collectionID string) *CollectionGr
 //
 // If a Collection Group Query would return a large number of documents, this
 // can help to subdivide the query to smaller working units that can be distributed.
+//
+// If the goal is to run the queries across processes or workers, it may be useful to use
+// `Query.ToProto` and `Query.FromProto` to serialize the query.
```

Comment:
This is good. I think it would be good to have a longer doc (here or elsewhere, not sure where the best place is, maybe the docs for the Query type?) that describes step-by-step how to run queries across workers, at least on a high level. Or if that best practices doc exists in the cloudsite firestore docs, we should link to it.

I guess the flow would be (for multiple worker machines): partition query, serialize to protos, send to workers, get result, deserialize. Or for threads on a single machine: partition query, spin up 1 goroutine per partition, run, collect results (no serialization/deserialization needed). Do I have that right?

Reply:
That is my thinking. if using goroutines I figure you just use the Query directly. The design calls to make sure we can serialize since this feature is mostly used for large collection groups, so beneficial to be able to partition. That said, I imagine for go, most users won't be doing this. I am working with another on creating user samples and solutions in the near future and I think, to your point, this would benefit from that